File: src/main/java/com/varun/classicalproblems/producerconsumer/Producer.java
package com.varun.classicalproblems.producerconsumer;

import java.util.UUID;

public class Producer implements Runnable {

  private final BoundedTaskQueue taskQueue;

  public Producer(BoundedTaskQueue taskQueue) {
    this.taskQueue = taskQueue;
  }

  @Override
  public void run() {
    while (true) {
      try {
        taskQueue.addTask(new Task(UUID.randomUUID().toString()));
      } catch (InterruptedException e) {
        e.printStackTrace();
      }
    }
  }
}

File: go/producerconsumer/model/producer.go
package model

import (
	"github.com/google/uuid"
)

type Producer struct {
	taskQueue *BoundedQueue
}

func (p *Producer) Produce() {
	for {
		p.taskQueue.AddTask(&Task{uuid.New().String()})
	}
}

func NewProducer(taskQueue *BoundedQueue) *Producer {
	return &Producer{
		taskQueue: taskQueue,
	}
}
